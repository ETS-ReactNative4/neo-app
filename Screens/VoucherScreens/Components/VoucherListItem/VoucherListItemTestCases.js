import React from "react";
import VoucherListItem from "./VoucherListItem";

const voucherItemsData = [
  {
    name: "Confirmation number",
    value: "SX236283624",
    info: "(19 Jun 2019 - 19 Jul 2019)"
  }
];

const VoucherListItemTestCases = [
  {
    title: "Voucher List Item",
    Component: (
      <VoucherListItem
        name={voucherItemsData[0].name}
        value={voucherItemsData[0].value}
      />
    )
  },
  {
    title: "Voucher List Item with info",
    Component: (
      <VoucherListItem
        name={voucherItemsData[0].name}
        value={voucherItemsData[0].value}
        info={voucherItemsData[0].info}
      />
    )
  },
  {
    title: "Voucher List Item with info Last",
    Component: (
      <VoucherListItem
        name={voucherItemsData[0].name}
        value={voucherItemsData[0].value}
        info={voucherItemsData[0].info}
        isLast={true}
      />
    )
  }
];

export default VoucherListItemTestCases;
