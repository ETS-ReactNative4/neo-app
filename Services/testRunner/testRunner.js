import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
