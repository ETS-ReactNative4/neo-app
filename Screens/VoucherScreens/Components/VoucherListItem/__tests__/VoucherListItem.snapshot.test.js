import VoucherListItemTestCases from "../VoucherListItemTestCases";
import testRunner from "../../../../../Services/testRunner/testRunner";

VoucherListItemTestCases.map(testRunner);
