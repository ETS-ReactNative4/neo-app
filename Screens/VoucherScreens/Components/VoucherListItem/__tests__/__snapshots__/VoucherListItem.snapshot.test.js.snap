// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voucher List Item 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 24,
      },
      Object {
        "borderBottomColor": "rgba(239, 239, 239, 1)",
        "borderBottomWidth": 1,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(119, 119, 119, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 17,
            "textAlign": "left",
          },
          Object {},
        ]
      }
    >
      Confirmation number
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-end",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(44, 47, 66, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 17,
            "textAlign": "right",
          },
          Object {},
        ]
      }
    >
      SX236283624
       
    </Text>
  </View>
</View>
`;

exports[`Voucher List Item with info 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 24,
      },
      Object {
        "borderBottomColor": "rgba(239, 239, 239, 1)",
        "borderBottomWidth": 1,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        },
        Object {},
      ]
    }
  >
    <Unknown
      style={
        Array [
          Object {
            "color": "rgba(119, 119, 119, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 17,
            "textAlign": "left",
          },
          Object {},
        ]
      }
    >
      Confirmation number
    </Unknown>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-end",
        },
        Object {},
      ]
    }
  >
    <Unknown
      style={
        Array [
          Object {
            "color": "rgba(44, 47, 66, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 17,
            "textAlign": "right",
          },
          Object {},
        ]
      }
    >
      SX236283624
       
    </Unknown>
    <Unknown
      style={
        Array [
          Object {
            "color": "rgba(170, 170, 170, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 17,
          },
          Object {},
        ]
      }
    >
      (19 Jun 2019 - 19 Jul 2019)
    </Unknown>
  </View>
</View>
`;

exports[`Voucher List Item with info Last 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 24,
      },
      Object {},
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        },
        Object {},
      ]
    }
  >
    <Unknown
      style={
        Array [
          Object {
            "color": "rgba(119, 119, 119, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 17,
            "textAlign": "left",
          },
          Object {},
        ]
      }
    >
      Confirmation number
    </Unknown>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-end",
        },
        Object {},
      ]
    }
  >
    <Unknown
      style={
        Array [
          Object {
            "color": "rgba(44, 47, 66, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 17,
            "textAlign": "right",
          },
          Object {},
        ]
      }
    >
      SX236283624
       
    </Unknown>
    <Unknown
      style={
        Array [
          Object {
            "color": "rgba(170, 170, 170, 1)",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 14,
            "lineHeight": 17,
          },
          Object {},
        ]
      }
    >
      (19 Jun 2019 - 19 Jul 2019)
    </Unknown>
  </View>
</View>
`;
